src/mcs/test/test_defmcs.jl

```diff
@@ -39,11 +45,44 @@ function print_result(m::Model, mcs::MonteCarloSimulation, trialnum::Int)
     println("$(ci.comp_id).E_Global: $value")
 end
 
-N = 100
+N = 1000
 
-generate_trials!(mcs, N, filename="/tmp/trialdata.csv")
+generate_trials!(mcs, N, filename="/tmp/Mimi/trialdata.csv")
 
 # Run trials 1:N, and save results to the indicated directory
 run_mcs(m, mcs, N, output_dir="/tmp/Mimi")
 
 # run_mcs(m, mcs, N, post_trial_func=print_result, output_dir="/tmp/Mimi")
+
+function show_E_Global(year::Int; bins=40)
+    df = @from i in E_Global begin
+             @where i.time == year
+             @select i
+             @collect DataFrame
+        end
+    histogram(df[:E_Global], bins=bins, 
+              title="Distribution of global emissions in $year",
+              xlabel="Emissions")
+end
+
+#
+# Test new loop capability
+#
+
+#
+# Save a pointer to the generated function in the mcs struct
+# Also save an optional context::Any that caller can use as needed
+#
+function my_loop_func(m::Model, mcs::MonteCarloSimulation,    # required args
+                      scen::Symbol, rate::Float64)            # user-defined args
+    # Do stuff with mcs and tuple values to set up model
+    println("scen:$scen, rate:$rate")
+end
+
+# Pass as a tuple of pairs
+run_mcs(m, mcs, 10;
+        output_dir="/tmp/Mimi",
+        loop_func=my_loop_func, 
+        loop_args=[:scen => [:low, :med, :high],
+                   :rate => [0.015, 0.03, 0.05]])
+ 
```

Comment:
All the test files seem to be in the wrong folder? None of them should be in the ``/src`` folder, they should all go in the test folder (or subfolders of that).

Reply:
Yeah, these aren't "proper" tests yet. I didn't want to move them to the test folder until I'd integrated them with the testing framework. I can do that next.